### PR TITLE
[IMPROVE][ENTERPRISE] Improve how micro services are loaded

### PR DESCRIPTION
**Pull request by ggazzo**

### CODE REVIEW

File: ee/server/NetworkBroker.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+const lifecycle: { [k: string]: string } = {
+	created: 'created',
+	started: 'started',
+	stopped: 'stopped',
+};
+
+const {
+	INTERNAL_SERVICES_ONLY = 'false',
+	// SERVICES_ALLOWED = '',
```
Comment (sampaiodiego) [inline ee/server/NetworkBroker.ts:23]: to remove `SERVICES_ALLOWED` you need to add the `internal` flag on every internal service currently missing the flag (like the `TeamService` for example)

File: ee/server/NetworkBroker.ts [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		try {
+			await this.broker.waitForServices(
+				method.split('.')[0],
+				this.isActionWhitelisted(method) ? waitForServicesWhitelistTimeout : waitForServicesTimeout,
+			);
+		} catch (err) {
+			console.error(err);
+		}
+
+		return this.call(method, data);
```
Comment (sampaiodiego) [inline ee/server/NetworkBroker.ts:98]: I noticed you're not using the `context` stored on `asyncLocalStorage` here anymore, any reason for that?

File: ee/server/NetworkBroker.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+		if (!this.isServiceAllowed(instance)) {
+			return;
+		}
+
+		const name = instance.getName();
+
+		if (!this.isServiceInternal(instance)) {
+			instance.onEvent('shutdown', async (services) => {
+				const service = services[name];
+				if (!service) {
```
Comment (sampaiodiego) [inline ee/server/NetworkBroker.ts:115]:
```suggestion
				if (!services[name]) {
```

File: ee/server/NetworkBroker.ts [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+			actions: {},
+			...(name !== 'license'
+				? {
+						dependencies: ['license'],
+				  }
+				: {}),
+			events: instance.getEvents().reduce<Record<string, Function>>((map, eventName) => {
+				map[eventName] = /^\$/.test(eventName)
+					? (data: Parameters<EventSignatures[typeof eventName]>): void => {
+							instance.emit(eventName, data);
```
Comment (sampaiodiego) [inline ee/server/NetworkBroker.ts:133]: any reason this is not `instance.emit(eventName, ...data)` anymore?

File: ee/app/license/server/license.internalService.ts [hunk 5]
```diff
@@ -26,6 +26,44 @@ class LicenseService extends ServiceClass implements ILicense {
 		onModule((licenseModule) => {
 			api.broadcast('license.module', licenseModule);
 		});
+
+		this.onEvent('$services.changed', async () => {
+			// if (hasModule('scalability')) {
```
Comment (sampaiodiego) [inline ee/app/license/server/license.internalService.ts:31]: should this be uncommented? (is that a real word? 😳 )

File: server/services/startup.ts [hunk 6]
```diff
@@ -1,28 +1,40 @@
...
-api.registerService(new LDAPService());
 api.registerService(new SAUMonitorService());
+api.registerService(new TeamService(db));
+api.registerService(new UiKitCoreApp());
+
+// if TRANSPORTER env var it means the process is running in micro services mode
+// in that case we don't need to register services that will run separately
+if (!process.env.TRANSPORTER?.match(/^(?:nats|TCP)/)) {
+	(async (): Promise<void> => {
+		const { Authorization } = await import('./authorization/service');
```
Comment (KevLehman) [inline server/services/startup.ts:36]: I'm curious 🤔 can this be a separate function?
Like:
```
api.importAndRegister(path: string, ...deps: any[])
```

That way, we won't have to import things here and the iife won't grow too big 😬

Comment (sampaiodiego) [reply server/services/startup.ts:36]: hmm.. interesting 🤔  I'll take a look and see if it works

Comment (sampaiodiego) [reply server/services/startup.ts:36]: this is definitely a good idea.. moleculer has the same feature to create/register services by using `broker.loadService('./file.js')` .. 

this is not something I'll provide on this PR, but I'll take notes to look at this in the future 👍 

thx again for bringing up this idea @KevLehman

**Review by sampaiodiego — DISMISSED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**